tests/codegen/mem-replace-direct-memcpy.rs

```diff
@@ -1,33 +0,0 @@
-// This test ensures that `mem::replace::<T>` only ever calls `@llvm.memcpy`
-// with `size_of::<T>()` as the size, and never goes through any wrapper that
-// may e.g. multiply `size_of::<T>()` with a variable "count" (which is only
-// known to be `1` after inlining).
-
-// compile-flags: -C no-prepopulate-passes -Zinline-mir=no
-// ignore-debug: the debug assertions get in the way
-
-#![crate_type = "lib"]
-
-pub fn replace_byte(dst: &mut u8, src: u8) -> u8 {
-    std::mem::replace(dst, src)
-}
```

Comment:
The "direct memcpy" name for this test no longer made sense, as it doesn't call `memcpy` any more.

What it was testing is subsumed by the new `tests/codegen/mem-replace-simple-type.rs` test below.